crates/nu-engine/src/documentation.rs

```diff
@@ -327,25 +423,40 @@ pub fn document_shape(shape: SyntaxShape) -> SyntaxShape {
 }
 
 pub fn get_flags_section<F>(
+    engine_state_opt: Option<&EngineState>,
```

Comment:
Under which circumstances should this or has this to be `None`?

Reply:
In the nu-plugin crate, `get_flags_section` gets called but there does not seem to be a way to get the engine state without creating one with all of its values [https://github.com/nushell/nushell/blob/main/crates/nu-plugin/src/plugin/mod.rs#L388](https://github.com/nushell/nushell/blob/main/crates/nu-plugin/src/plugin/mod.rs#L388).

Reply:
Ah that is unfortunate, but yeah simplifying the plugin side of things is probably too much for this PR.
We can keep that as a todo for the future.